File: chor-extraction/src/main/java/ast/sp/nodes/ProcedureInvocation.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;

/**
 * Created by lara on 04/04/17.
 */
public class ProcedureInvocation implements Behaviour {
    public final String procedure;

    public ProcedureInvocation(String procedure) {
        this.procedure = procedure;
    }

    public String getProcedure() {
        return procedure;
    }
}

File: pe/src/main/kotlin/ast/cc/nodes/ProcedureInvocation.kt
package ast.cc.nodes

import ast.cc.interfaces.CCVisitor
import ast.cc.interfaces.ChoreographyBody
import java.util.*

data class ProcedureInvocation(val procedure: String) : ChoreographyBody() {
    override fun <T> accept(visitor: CCVisitor<T>): T = visitor.visit(this)

    override fun toString() = procedure
}
